flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliClient.java

```diff
@@ -253,7 +253,13 @@ public boolean submitUpdate(String statement) {
 	// --------------------------------------------------------------------------------------------
 
 	private Optional<SqlCommandCall> parseCommand(String line) {
-		final Optional<SqlCommandCall> parsedLine = SqlCommandParser.parse(executor.getSqlParser(sessionId), line);
+		final Optional<SqlCommandCall> parsedLine;
+		try {
+			parsedLine = SqlCommandParser.parse(executor.getSqlParser(sessionId), line);
+		} catch (SqlExecutionException e) {
+			printExecutionException(e);
+			return Optional.empty();
+		}
 		if (!parsedLine.isPresent()) {
 			printError(CliStrings.MESSAGE_UNKNOWN_SQL);
```

Comment:
We may never run into this code branch now ~

Reply:
I think we should keep this. because when TableEnvironment supports new operation but forgets to update SqlCommandParser, we should print error here or throw exception in SqlCommandParser

Reply:
> I think we should keep this. because when TableEnvironment supports new operation but forgets to update SqlCommandParser
I think we should never do this, either to refactor the code to keep SQL_CLI synced with table environment or keep the logic clean.